### PR TITLE
git-archive: add page

### PR DESCRIPTION
**Pull request by glenacota**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

Related to issue #3953

### CODE REVIEW

File: pages/common/git-archive.md
Comment (navarroaxel) [inline pages/common/git-archive.md:8]: We prefer the long argumento format:
```suggestion
`git archive --verbose --output={{path/to/file.tar}} HEAD`
```

File: pages/common/git-archive.md
Comment (navarroaxel) [inline pages/common/git-archive.md:12]:
```suggestion
`git archive --output={{path/to/file.tar}} HEAD:{{path/to/directory}}`
```

File: pages/common/git-archive.md
Comment (navarroaxel) [inline pages/common/git-archive.md:16]:
```suggestion
`git archive --output={{path/to/file.tar}} --prefix={{prefix}} HEAD:{{path/to/directory}}`
```

File: pages/common/git-archive.md
Comment (navarroaxel) [inline pages/common/git-archive.md:14]: I'm not sure about this description, maybe the example is clearer if we say `preffix/` to include all the files inside a specific folder

File: pages/common/git-archive.md
Comment (navarroaxel) [inline pages/common/git-archive.md:24]: Is this a typo? or `git` change the file type to `zip` using the extension?
```suggestion
`git archive --output={{path/to/file.tar}} {{branch_name}}`
```

File: pages/common/git-archive.md
Comment (navarroaxel) [inline pages/common/git-archive.md:6]: Maybe an extra example (at the first position) saying that the output is written to the `stdout` is better.

**Review by navarroaxel — CHANGES_REQUESTED** [review]
Hey @glenacota, thanks for your contribution. I left some comments.

File: pages/common/git-archive.md
Comment (glenacota) [reply pages/common/git-archive.md:8]: Will use the long argument :)
As for the file name, I used the same convention used for the `tar` command, e.g., `tar cf {{target.tar}} {{file1}} {{file2}} {{file3}}`

File: pages/common/git-archive.md
Comment (glenacota) [reply pages/common/git-archive.md:24]: yes, it's inferred. From the docs:
> If this option is not given, and the output file is specified, the format is inferred from the filename if possible (e.g. writing to "foo.zip" makes the output to be in the zip format)

**Comment by glenacota** [comment]
Thanks for the review @navarroaxel ! Please have a look at the latest changes. I only kept `{{target.tar}}` instead of `{{path/to/target.tar}}` because (1) it's also `{{target.tar}}` for the `tar` command, (2) it keeps the example shorter. What do you think?

**Review by owenvoke — APPROVED** [review]
LGTM after @navarroaxel's comments have been addressed. 👍🏻

**Comment by navarroaxel** [comment]
Ref: #3953

**Comment by glenacota** [comment]
Thanks, @owenvoke! @navarroaxel, there is still 1 requested change pending, but I'm not sure what it refers to. Is it the `target.tar` one? Is it your final decision to stick to go for `path/to/file.tar`, and then update the `tar` page consistently afterwards?

File: pages/common/git-archive.md
Comment (sbrl) [reply pages/common/git-archive.md]: Hey! Yeah, filenames can be an awkward one. Generally, the rule of thumb is that we prefer including the `path/to/` prefix. However, if you've got a number of files in a row (e.g. see the page for the `diff` command), then we drop the `path/to/` prefix to avoid the example from being unnecessarily verbose. In this case though I think we can include it.

**Comment by sbrl** [comment]
Hey! I've left a comment about the filename issue above.

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**
